cli/azd/pkg/spin/run.go

```diff
@@ -2,13 +2,31 @@ package spin
 
 import (
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/theckman/yacspin"
 )
 
+// SpinnerUpdater contains callbacks for updating the spinner.
+type SpinnerUpdater struct {
```

Comment:
Consider moving the Spinner updates into a smaller separate PR.  Would be good to have some unit tests here as well.

Reply:
Split into #166